akv-plugin/proto/key.cs

```diff
@@ -0,0 +1,107 @@
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Notation.Plugin.Proto
+{
+    /// <summary>
+    /// KeySpec class.
+    /// </summary>
+    public class KeySpec
+    {
+        public string Type { get; }
+        public int Size { get; }
+        public KeySpec(string type, int size)
+        {
+            if (string.IsNullOrEmpty(type))
+            {
+                throw new ArgumentNullException(nameof(type), "Type must not be null or empty");
+            }
+
+            Type = type;
+            Size = size;
+        }
+    }
+
+    /// <summary>
+    /// KeySpecUtils class.
+    /// </summary>
+    public static class KeySpecUtils
+    {
+        public const string RSA2048 = "RSA-2048";
+        public const string RSA3072 = "RSA-3072";
+        public const string RSA4096 = "RSA-4096";
+        public const string EC256 = "EC-256";
+        public const string EC384 = "EC-384";
+        public const string EC521 = "EC-521";
+
+        /// <summary>
+        /// Extracts the key spec from the certificate.
+        /// </summary>
+        public static KeySpec ExtractKeySpec(X509Certificate2 signingCert)
+        {
+            RSA? rsaKey = signingCert.GetRSAPublicKey();
+            ECDsa? ecdsaKey = signingCert.GetECDsaPublicKey();
+
+            if (rsaKey != null)
+            {
+                int bitSize = rsaKey.KeySize;
+
+                if (bitSize == 2048 || bitSize == 3072 || bitSize == 4096)
+                {
+                    return new KeySpec("RSA", bitSize);
+                }
+
+                throw new ValidationException($"RSA key size {bitSize} bits is not supported");
+            }
+            else if (ecdsaKey != null)
+            {
+                int bitSize = ecdsaKey.KeySize;
+
+                if (bitSize == 256 || bitSize == 384 || bitSize == 521)
+                {
+                    return new KeySpec("EC", bitSize);
+                }
+
+                throw new ValidationException($"ECDSA key size {bitSize} bits is not supported");
+            }
+
+            throw new ValidationException("Unsupported public key type");
```

Comment:
Can we have other better way to determine the public key?

Reply:
We can get the key by `signingCert.PublicKey.Key` but `PublicKey.Key` is [obsolete](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.publickey.key?view=net-8.0). It asks me to use the appropriate method to get the public key, such as `GetRSAPublicKey`.

So I optimize the code based on current logic.